### PR TITLE
Review

### PR DESCRIPTION
**Pull request by vasilevnikitad**

### CODE REVIEW

File: ip_filter.hpp [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      template<typename Container>
+      inline auto get_filtered_ip(Container &&in)
+      {
+        std::remove_reference_t<decltype(in)> out;
+
+        std::copy_if(std::begin(in),
+                     std::end(in),
+                     std::back_insert_iterator<decltype(out)>(out),
+                     [&](ip const &ip_addr){ return is_valid(ip_addr); });
+
```
Comment (SergeiNA) [inline ip_filter.hpp]: Complexity- O(N) .  Have any idea how to decreace to ~log(N) ?

Comment (vasilevnikitad) [reply ip_filter.hpp]: Good question!

I can sort it and then return just a range of elements!

Thank you!

File: main.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    ip_pool.emplace_back(line.substr(0, line.find('\t')));
+
+  std::sort(std::begin(ip_pool), std::end(ip_pool), std::greater<ip_filter::ip>());
+
+  apply_filters(ip_pool,
+      [](ip_filter::ip const &ip){return true;},
+      [](ip_filter::ip const &ip){return ip[0] == 1;},
+      [](ip_filter::ip const &ip){return ip[0] == 46 && ip[1] == 70;},
+      [](ip_filter::ip const &ip){return ip[0] == 46 || ip[1] == 46 ||
+                                         ip[2] == 46 || ip[3] == 46;}
```
Comment (SergeiNA) [inline main.cpp]: Cute smart! But  don't you think it's far distant from:
filter(1)
filter(46,70)
filter_any(46) <--- especially for this!!!! Checking every numer in main func loks odd.
Why don't you write some wrapper?

Comment (vasilevnikitad) [reply main.cpp]: >Checking every numer in main func loks odd.

I will add iterator support and then will use std::any_of in this case

>Why don't you write some wrapper?

Well,  maybe it's a good idea in some way. But can you understand what it means
`filter(1)` or `filter(46, 70) from code? As for me, this weird numbers tells nothing. 
That's why lambdas w/o wrappers are better in this case.

**Pull request merged**